src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java

```diff
@@ -250,14 +270,27 @@ public void userEventTriggered(ChannelHandlerContext context, Object evt) throws
         return new ChannelInitializer<Channel>() {
             @Override
             protected void initChannel(Channel channel) throws Exception {
-                if(useSsl) {
+                if (useSsl) {
                     channel.pipeline().addLast(newSslInitiator());
                 }
                 channel.pipeline().addLast(new LengthFieldBasedFrameDecoder(Integer.MAX_VALUE, 1, 4, -4, 0, true));
                 channel.pipeline().addLast(new ByteBufMessageDecoder());
                 channel.pipeline().addLast(new ByteBufMessageEncoder());
                 channel.pipeline().addLast(newProtocolHandler());
                 channel.pipeline().addLast(onActive);
+                channel.pipeline().addLast(newExceptionHandler());
+            }
+        };
+    }
+
+    private ChannelInboundHandlerAdapter newExceptionHandler() {
+        return new ChannelInboundHandlerAdapter() {
+            @Override
+            public void exceptionCaught(ChannelHandlerContext context, Throwable cause) throws Exception {
+                Collection<Subscriber<? super Message>> unsubscribe = subscribers;
```

Comment:
This looks fishy. As removeAll is called with itself, unsubscribe is always empty and no onError calls are performed. The original version looks wrong as well, this should probably be something like:

```java
if(!subscribers.isEmpty()) {
    Collection<Subscriber<? super Message>> unsubscribed = new ArrayList<>(subscribers);
    subscribers.clear(); // clear subs before onError to prevent recursive invocation
    unsubscribed.forEach(subscriber -> {
        try {
            subscriber.onError(cause)
        } catch(Exception ex) { /* ignored */ }
    });
}
```

Reply:
better you can extract try{ } catch to separate method and also

```if(!subscribers.isEmpty())```
is not needed. forEach on empty list is harmless.